test/helpers/role_organisations_helper_test.rb

```diff
@@ -0,0 +1,29 @@
+require "test_helper"
+
+class RoleOrganisationsHelperTest < ActionView::TestCase
+  context "#options_for_your_organisation_select" do
+    setup do
+      @user_organisation = create(:organisation, name: "User Organisation", abbreviation: "UO")
+      @other_organisation = create(:organisation, name: "Other Organisation")
+      @user = create(:admin_user, organisation: @user_organisation)
+    end
+
+    should "use the name with abbreviation for the organisation as the option text" do
+      assert(options_for_your_organisation_select(@user).detect { |o| o[:text] == @user_organisation.name_with_abbreviation })
+      assert(options_for_your_organisation_select(@user).detect { |o| o[:text] == @other_organisation.name_with_abbreviation })
+    end
+
+    should "use the id of the organisation as the value" do
+      assert(options_for_your_organisation_select(@user).detect { |o| o[:value] == @user_organisation.id })
+      assert(options_for_your_organisation_select(@user).detect { |o| o[:value] == @other_organisation.id })
+    end
+
+    should "select the user's organisation" do
+      user_organisation_option = options_for_your_organisation_select(@user).detect { |o| o[:value] == @user_organisation.id }
+      other_organisation_option = options_for_your_organisation_select(@user).detect { |o| o[:value] == @other_organisation.id }
+
+      assert user_organisation_option[:selected]
+      assert_not other_organisation_option[:selected]
+    end
```

Comment:
Subjective: I find the assertions in these tests hard to read. I think it's partly because they are quite long lines but also if feels as if there's two things going on - (1) invoking the method under test; and (2) digging into the return value to make the assertion. I think I'd find them more readable if they were split up along those lines, e.g.

```
    should "use the name with abbreviation for the organisation as the option text" do
      options = options_for_your_organisation_select(@user)

      assert options.detect { |o| o[:text] == @user_organisation.name_with_abbreviation }
      assert options.detect { |o| o[:text] == @other_organisation.name_with_abbreviation }
    end
```

Note that I've also removed some parens from the assertion which I think are redundant.

Thinking about it a bit more, I think I might be inclined to just assert the whole structure in a single test like I did here:

https://github.com/alphagov/signon/blob/3f25de1c9796a1c5584dc4267c9649965c9cf5c2/test/helpers/users_helper_test.rb#L77-L113

Reply:
Addressed in 44bba6a0